### DISCUSSION TITLE
Should useMutation provide an option to keep the keepPreviousData?

### DISCUSSION THREAD

**Discussion by ericchernuka** [TanStack/query#1642] [Ideas]
Thought I'd ask this before investigating a PR. I have a use case where we have preview functionality within our app to show rendered content. We useMutation to get the API to provide us an updated version of the preview, but upon update the data is cleared when executing the mutation again. The workaround currently is to have a ref above that keeps track of the rendered preview and does not clear it between executions to avoid the content flashing when it's being recreated.

Does anyone have a need for keeping the previous mutate data between executions? I realize one aspect that would need to be figured out would be what happens if an error occurs (keep or clear data).

**Comment by TkDodo** [comment]
If you have a `query` that corresponds to the mutation, you can update / invalidate that query and always display the query rather than the mutation result.

**Reply by ericchernuka** [reply]
That's a good idea, but sadly its all computed on the fly due to the level of personalization.

**Comment by amar-josh** [comment]
Hi... @ericchernuka have you found any solution for this problem. I am also facing the same issue

**Reply by ericchernuka** [reply]
I haven't and sadly no longer work on that project. I'll keep this thread posted if friends on that team solve it in a different way.

**Reply by amar-josh** [reply]
Ohhh... 
Thanks for the Quick reply.

**Reply by joyfulwang** [reply]
Hi, @amar-josh and @ericchernuka. Were either of you or your teams able to find a solution for the issue? I'm currently encountering a similar one

Or at the very least, would much appreciate seeing the ref workaround that has been implemented
